taca/analysis/analysis.py

```diff
@@ -91,9 +91,40 @@ def transfer_run(run, analysis=True):
         tsv_writer.writerow([os.path.basename(run), str(datetime.now())])
     os.remove(os.path.join(run, 'transferring'))
 
+    #Now, let's move the run to nosync
+    archive_run(run)
+
     if analysis:
         trigger_analysis(run)
 
+def archive_run(run)
+    
+    rppath=os.path.join(run, 'runParameters.xml')
+    try:
+        rp=XTenRunParametersParser(os.path.join(run, 'runParameters.xml'))
+    except OSError:
+        logger.warn("Cannot find the runParameters.xml file at {}. This is quite unexpected. please archive the run {} manually".format(rppath, run))
+    else:
+        try:
+            #Works for recent control software
+            runtype=rp.data.["Setup"].get("Application Name")
+        except KeyError :
+            #should work for ancient control software
+            runtype=rp.data.get("Application Name")
+
+        if "HiSeq X" in runtype:
```

Comment:
that kinda defeats the point of having them in the config file. We could
change that to a key/value pair :
HiSeq : <path>
MiSeq : <path>
HiseqX: <path>

what do you think ?


On Tue, Jun 9, 2015 at 5:04 PM, Guillermo Carrasco <notifications@github.com
> wrote:

> In taca/analysis/analysis.py
> <https://github.com/SciLifeLab/TACA/pull/130#discussion_r32022763>:
>
> > +def archive_run(run)
> > +
> > +    rppath=os.path.join(run, 'runParameters.xml')
> > +    try:
> > +        rp=XTenRunParametersParser(os.path.join(run, 'runParameters.xml'))
> > +    except OSError:
> > +        logger.warn("Cannot find the runParameters.xml file at {}. This is quite unexpected. please archive the run {} manually".format(rppath, run))
> > +    else:
> > +        try:
> > +            #Works for recent control software
> > +            runtype=rp.data.["Setup"].get("Application Name")
> > +        except KeyError :
> > +            #should work for ancient control software
> > +            runtype=rp.data.get("Application Name")
> > +
> > +        if "HiSeq X" in runtype:
>
> hmmm you're assuming here an implicit order of archive_dir in the
> configuration paths, which I wouldn't do... these config files are changed
> quite often. What I did is try to find the run in one of the data_dirs
> and the append the nosync to the one that has the run.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/TACA/pull/130/files#r32022763>.
>